### PR TITLE
[#3832] Hide tab bar on Android when keyboard visible

### LINKED ISSUE #3832: Hide tab bar when editing userpick in the profile
### User Story

As a user, I don't want to see a tab bar while editing the photo so that I could be fully focused on that process.

### Description

*Type*: Bug

*Summary*: Fun Android bug that keeps showing a tab bar above keyboard while a user is editing userpick or name. Tab bar should be hidden. Check iOS build for reference.  

#### Expected behavior
screenshot 2018-04-11 11 10 45

#### Actual behavior
image 2018-04-11 11 08 55

### Reproduction
[comment]: # (Describe how we can replicate the bug step by step.)

- Open Status
- Open profile tab
- Tap Edit in the right top corner

### Additional Information
* Status version: nightly 10.05.2018
* Operating System: Android

### PR DESCRIPTION
**Pull request by kwingram25**
Fixes #3832

### Summary:
Hides tab bar on Android when keyboard is visible (i.e. editing user profile)

### Steps to test:
- Open Status
- Create account, login, Profile, tap Edit
- Tab bar should hide while keyboard is up

Status: Ready

### CODE REVIEW

File: src/status_im/ui/screens/main_tabs/views.cljs [hunk 1]
```diff
@@ -65,7 +65,8 @@
      ^{:key view-id} [tab view-id content (= view-id current-view-id) accessibility-label count-subscription])])
 
 (views/defview main-tabs []
-  (views/letsubs [view-id [:get :view-id]]
+  (views/letsubs [view-id          [:get :view-id]
+                  tab-bar-visible? [:get :tab-bar-visible?]]
```
Comment (cammellos) [inline src/status_im/ui/screens/main_tabs/views.cljs]: We are moving away from the pattern of using:

`[:get :something]`

changing them to:

`[:get-something]`

Could you please create a new subscription for this event? Similarly for `[:set :something]`

File: src/status_im/android/core.cljs [hunk 2]
```diff
@@ -44,13 +44,16 @@
...
                          (let [h (.. e -endCoordinates -height)]
+                           (dispatch [:set :tab-bar-visible? false])
                            (when-not (= h @keyboard-height)
                              (dispatch [:set :keyboard-height h])
                              (dispatch [:set :keyboard-max-height h])))))
          (.addListener react/keyboard
                        "keyboardDidHide"
-                       #(when-not (= 0 @keyboard-height)
-                          (dispatch [:set :keyboard-height 0])))
+                       (fn [e]
```
Comment (cammellos) [inline src/status_im/android/core.cljs]: Seems like we are not using the `e` parameter, could you please change this to `_` ?

File: src/status_im/ui/screens/main_tabs/views.cljs (hunk 1 above)
Comment (cammellos) [reply src/status_im/ui/screens/main_tabs/views.cljs]: We could use these events:

`hide-tab-bar`
`show-tab-bar`

and the subscription:

`:tab-bar-visible?`

File: src/status_im/android/core.cljs [hunk 3]
```diff
@@ -44,13 +44,16 @@
...
+                         (dispatch [:set :tab-bar-visible? true])
+                         (when-not (= 0 @keyboard-height)
```
Comment (dmitryn) [inline src/status_im/android/core.cljs]: `(zero? @keyboard-height)`

**Review by cammellos — APPROVED** [review]

**Review by dmitryn — APPROVED** [review]

**Comment by Serhy** [comment]
Builds https://i.diawi.com/nE7KCH (Android) and https://i.diawi.com/wrpab7 (iOS) tested on Android 7.0, 7.1, 8.0, 6.0, and iOS 11.2.5. 
Looks good! No bottom tab bar while editing username. 

<blockquote><div><strong><a href="https://i.diawi.com/nE7KCH">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://i.diawi.com/wrpab7">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
Thanks for contribution @kwingram25!

**Pull request merged**